bin/weewx/imagegenerator.py

```diff
@@ -190,12 +190,23 @@ def genImages(self, gen_ts):
                     # Now its time to find and hit the database:
                     binding = line_options['data_binding']
                     db_manager = self.db_binder.get_manager(binding)
+                    # we need to pass the line options and plotgen_ts to our xtype
+                    # first get a copy of line_options
+                    option_dict = dict(line_options)
+                    # but we need to pop off aggregate_type and
+                    # aggregate_interval as they are used as explicit arguments
+                    # in our xtypes call
+                    option_dict.pop('aggregate_type', None)
```

Comment:
Don't see why you need to pop `aggregate_type` and `aggregate_interval`. Just remove them from the call to `get_series()`. The `**option_dict` would expand to include them.

Reply:
Maybe I am doing something wrong but when I leave them in I get a `TypeError: get_series() got multiple values for keyword argument 'aggregate_type'`